packages/outstatic/src/utils/metadata/load.ts

```diff
@@ -0,0 +1,154 @@
+            for (const [s, dir] of Object.entries(obj)) {
+              next = next.thenBy(s, { direction: dir >= 0 ? 'asc' : 'desc' })
+            }
+            return next
+          }
+          const applyString = (s: string, tb: IThenBy<any>) => {
+            return tb.thenBy(s)
+          }
+
+          if (typeof sort === 'number' || typeof sort === 'symbol') {
+            throw new Error(
+              'sort must be either an array of fields or an object containing field keys and sort directions'
+            )
+          }
+
+          if (typeof sort === 'string') {
+            subset.sort(firstBy<any>(sort, { direction: 'asc' }))
+            return api
+          }
+
+          if (!Array.isArray(sort)) {
+            let fn = firstBy<any>('\x00')
+            fn = applyKeys(sort, fn)
+            subset.sort(fn)
+            return api
+          }
+
+          // array
+          let fn: ReturnType<typeof firstBy<any>> | undefined
+          for (const entry of sort) {
+            if (!fn) {
+              if (typeof entry === 'string') {
+                fn = firstBy<any>(entry, { direction: 'asc' })
+                continue
+              } else {
+                fn = firstBy<any>('\x00') // placeholder
+              }
+            }
+
+            if (!fn) {
+              throw new Error('Unreachable condition: !fn')
+            }
+
+            if (typeof entry === 'string') {
+              fn = applyString(entry, fn)
+            } else if (typeof entry === 'number' || typeof entry === 'symbol') {
+              throw new Error(
+                'sort must be either an array of fields or an object containing field keys and sort directions'
+              )
+            } else {
+              fn = applyKeys(entry, fn)
+            }
+          }
+
+          if (fn) {
+            subset.sort(fn)
+          }
+          return api
+        },
+
+        project: (projection) => {
+          prj = projection
+          return api
+        },
+
+        skip: (skip) => {
+          skp = skip
+          return api
+        },
+
+        limit: (limit) => {
+          lmt = limit
+          return api
+        },
+
+        first: async () => {
+          const arr = await api.toArray()
+          return arr?.[0]
+        },
+
+        toArray: async () => {
+          // narrow down to smallest result set
+          const copied = (
+            JSON.parse(JSON.stringify(subset)) as OutstaticSchema<TSchema>[]
+          ).slice(skp, lmt)
+
+          const finalProjection = Array.isArray(prj) ? prj : Object.keys(prj)
+
+          // check projections and load content
+          const projected = await Promise.all(
+            copied.map(async (m) => {
+              if (
+                finalProjection.length === 0 ||
+                finalProjection.includes('content')
+              ) {
+                const buf = await readFile(
+                  resolve(CONTENT_PATH, `./${m.__outstatic.path}`)
```

Comment:
The error I've mentioned [here](https://github.com/avitorio/outstatic/pull/72/files#r1059684582), actually is triggered by this line. As you've mentioned the `__outstatic.path` has the full path (including monorepo and content path) so appending it to `CONTENT_PATH` creates a huge path. I don't know how this is working for you 😅

I've fixed it with regex on my end:
```
const regex = /\/[^\/]+\/[^\/]+$/
const path = m.__outstatic.path.match(regex)![0]
const buf = await readFile(
  resolve(join(CONTENT_PATH, `${path}`))
)
 ```
 
This only pulls the filename and collection from `m.__outstatic.path`, ie: `/pages/home.md`, not sure how future proof this is.

Perhaps the better option would be to make sure the metadata db only saves the path relative to the content folder. Basically, treat the GitHub path response before storing it on the db.

Reply:
Awesome. Taken that as a TODO. Can see rationale for why `__outstatic.path` has the full path here: https://github.com/avitorio/outstatic/pull/72#discussion_r1060200301

I don't think we want to introduce `NEXT_PUBLIC_*` env vars unless we have to, so it probably makes sense to resolve this inside of `toArray()` as you've done here

Reply:
Sounds good. I agree, introducing new env vars, especially public ones, feels like it's too much for now. Let's treat the path with regex and see if it holds for now. :)